types/smooch/index.d.ts

```diff
@@ -0,0 +1,954 @@
+    }
+
+    interface User {
+        id: string;
+        externalId: string;
+        signedUpAt: string;
+        hasPaymendInfo: boolean;
+        givenName: string;
+        surname: string;
+        email: string;
+        avatarUrl: string;
+        metadata: {
+            justGotUpdated: boolean;
+        };
+    }
+
+    interface ConversationParticipant {
+        id: string;
+        userId: string;
+        unreadCount: number;
+        lastRead: number;
+    }
+
+    interface Message {
+        role: 'user' | 'business';
+        userId: string;
+        displayName: string;
+        id: string;
+        type: 'text' | 'formResponse' | 'form';
+        received: number;
+        text: string;
+        source: MessageSource;
+    }
+
+    interface SimpleMessage {
+        type: string;
+        text: string;
+    }
+
+    interface MessageSource {
+        type: 'web' | 'slack';
+        id: string;
+        integrationId: string;
+    }
+
+    interface Conversation {
+        id: string;
+        lastUpdatedAt: number;
+        businessLastRead: number;
+        description: string;
+        displayName: string;
+        iconUrl: string;
+        type: 'sdkGroup' | string;
+        participants: ConversationParticipant[];
+        metadata: {};
+        messages: Message[];
+    }
+
+    interface Status {
+        status: 'resolved' | 'pending' | string;
+    }
+
+    interface ConversationData {
+        conversation: Conversation;
+    }
+
+    /**
+     * Smooch allows you to set a delegate to receive callbacks when important changes happen in the conversation.
+     *
+     * To set a delegate, pass the `delegate` parameter in to [init options](#Smooch::init), or use the [setDelegate](#Smooch::setDelegate) method. The `delegate` object may optionally contain
+     * `beforeDisplay`, `beforeSend`, `beforePostbackSend` and `onInvalidAuth` delegate functions.
+     * Passing `delegate` as part of `init` options is the preferred method. The `setDelegate` method can be used to change or remove delegate behaviors after a conversation has been initialized.
+     *
+     * A `data` object is passed down with all the delegate events except `onInvalidAuth`. This is a read-only object containing a truncated version of the conversation associated with the event.
+     *
+     * `beforeSend` delegate will apply to the `formResponse` message sent when a [Prechat Capture](https://docs.smooch.io/guide/web-messenger/#prechat-capture) form is completed.
+     */
+    interface Delegate {
+        beforeDisplay?: (message: Message, data: ConversationData) => Message | null;
+        beforeSend?: (message: Message, data: ConversationData) => Message | null;
+        beforePostbackSend?: (postback: unknown, data: ConversationData) => unknown;
+        onInvalidAuth?: () => string | Promise<string>;
+    }
+}
+
+type Nullable<T> = { [P in keyof T]: T[P] | null };
+
+declare const smooch: typeof Smooch;
+
+export as namespace Smooch;
```

Comment:
Looking at the JS on unpkg, I don’t actually see a UMD wrapper—it looks like just a CJS `module.exports`, which would mean you can’t use as a global just by including a script tag and you should drop this line.

Reply:
> Looking at the JS on unpkg, I don’t actually see a UMD wrapper

yes, this is what we wrote, this is not an UMD wrapper. We just discussed how to approach a situation, if someone writes a non-module code, adding a script tag to the page, which still can find DT definition useful, say in VSCode:
The 'Smooch' object, (net yet unitialized) become available in global script:
```
Smooch
{init: ƒ, on: ƒ, render: ƒ, destroy: ƒ}
```